### PR TITLE
allow arbitrary strings for ssl_verify option.

### PR DESCRIPTION
**Pull request by groutr**
Addresses https://github.com/conda/conda/issues/1387

### CODE REVIEW

**Comment by asmeurer** [comment]
One issue I see with this is that it will break backwards compatibility with `conda config --set ssl_verify false`.

File: conda/config.py [hunk 1]
```diff
@@ -331,7 +331,15 @@ def get_proxy_servers():
 disallow = set(rc.get('disallow', []))
 # packages which are added to a newly created environment by default
 create_default_packages = list(rc.get('create_default_packages', []))
-ssl_verify = bool(rc.get('ssl_verify', True))
+
+# ssl_verify can be a boolean value or a filename string
+ssl_verify = rc.get('ssl_verify', 'True')
+_v = ssl_verify.lower()
+if _v in {'on', 'true', 'yes'}:
```
Comment (asmeurer) [inline conda/config.py]: This logic is not necessary. YAML already takes care of this.

**Comment by groutr** [comment]
The newest changes allow setting of either boolean or string keys.  This should restore backwards compatibility.

**Comment by groutr** [comment]
Are there any other changes that should be made regarding this feature?

**Comment by asmeurer** [comment]
Great. Maybe a test should be added to the config tests. Otherwise, looks good assuming the tests pass.

File: conda/cli/main_config.py [hunk 2]
```diff
@@ -100,22 +100,11 @@ def __init__(self, reason):
...
-            ret = ret or isinstance(yaml.load(other), bool)
-
-        return ret
-
+class SingleValueKey(common.Completer):
     def _get_items(self):
-        return config.rc_bool_keys + ['yes', 'no', 'on', 'off', 'true', 'false']
+        return config.rc_bool_keys + \
+               config.rc_string_keys + \
+               ['yes', 'no', 'on', 'off', 'true', 'false']
```
Comment (groutr) [inline conda/cli/main_config.py]: I'm not convinced we should be auto completing values for keys.

Comment (asmeurer) [inline conda/cli/main_config.py]: I think we should. It's helpful.

**Comment by asmeurer** [comment]
@jasongrout were you going to test this?

File: conda/fetch.py [hunk 3]
```diff
@@ -145,6 +145,11 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
         log.debug(msg)
         raise RuntimeError(msg)
 
+    except requests.exceptions.SSLError as e:
```
Comment (asmeurer) [inline conda/fetch.py]: SSLError subclasses from ConnectionError, meaning that if somehow a 407 gives this exception the https proxy handling will break.

**Comment by asmeurer** [comment]
It would be great to test this on a proxy if possible.

**Comment by jasongrout** [comment]
@asmeurer - is there an easy way to pull these changes into a local install to test them?  I'm just running a stock anaconda distribution right now.

**Comment by jasongrout** [comment]
Never mind, I see your comment at https://github.com/conda/conda/issues/1387#issuecomment-123459275

**Comment by jasongrout** [comment]
This change seems to work great! Thanks!

File: conda/fetch.py (hunk 3 above)
Comment (groutr) [inline conda/fetch.py]: Maybe if we check for a 407, re-raising if it is.

Comment (asmeurer) [inline conda/fetch.py]: Reraising an exception doesn't cause it to be caught by further `except` blocks.

**Pull request merged**